apps/test/unit/lib/kits/maker/boards/microBit/CapacitiveTouchSensorTest.js

```diff
@@ -53,26 +58,29 @@ describe('CapacitiveTouchSensor', function() {
       // Seed the channel with 'connected' data
       boardClient.analogChannel[testPin] = 450;
 
-      setInterval(() => {
-        expect(emitSpy).to.have.been.calledOnce;
-        expect(emitSpy).to.have.been.calledWith('down');
-      }, 50);
+      // Wait for readSensorTimer to finish
+      await new Promise(resolve => setInterval(resolve, 50));
+
+      expect(emitSpy).to.have.been.calledOnce;
+      expect(emitSpy).to.have.been.calledWith('down');
     });
 
-    it('emits the up event when it receives a low enough reading', () => {
+    it('emits the up event when it receives a low enough reading', async () => {
       boardClient.receivedAnalogUpdate();
 
       // Seed the channel with 'connected' data
       boardClient.analogChannel[testPin] = 450;
       boardClient.receivedAnalogUpdate();
+      sensor.connected = true;
```

Comment:
i'm not sure that the original test was actually passing; it may have just finished executing before it failed due to the previously-async nature of the test. when i checked the code that this test exercises ([here](https://github.com/code-dot-org/code-dot-org/blob/4ef9a58c4bebb5c3d45d7d3a353419c1f3dd3cca/apps/src/lib/kits/maker/boards/microBit/CapacitiveTouchSensor.js#L35-L55)), the value of `this.connected` was always false, but setting it to true here did the trick